pkg/api/norman/customization/cluster/actions_key.go

```diff
@@ -0,0 +1,71 @@
+package cluster
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/api/access"
+	"github.com/rancher/norman/types"
+	v3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	v3client "github.com/rancher/rancher/pkg/client/generated/management/v3"
+	"github.com/rancher/rancher/pkg/controllers/management/etcdbackup"
+	mgmtv3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (a ActionHandler) RotateEncryptionKey(actionName string, action *types.Action, apiContext *types.APIContext) error {
+	response := map[string]interface{}{
+		"type": v3client.RotateEncryptionKeyOutputType,
+		v3client.RotateEncryptionKeyOutputFieldMessage: "etcd backup created, rotating encryption key",
+	}
+
+	var mgmtCluster mgmtv3.Cluster
+	if err := access.ByID(apiContext, apiContext.Version, apiContext.Type, apiContext.ID, &mgmtCluster); err != nil {
+		response[v3client.RotateEncryptionKeyOutputFieldMessage] = "cluster does not exist"
+		apiContext.WriteResponse(http.StatusBadRequest, response)
+		return errors.Wrapf(err, "failed to get cluster by ID %s", apiContext.ID)
+	}
+
+	cluster, err := a.ClusterClient.Get(apiContext.ID, v1.GetOptions{})
+	if err != nil {
+		response[v3client.RotateEncryptionKeyOutputFieldMessage] = "cluster does not exist"
+		apiContext.WriteResponse(http.StatusBadRequest, response)
+		return errors.Wrapf(err, "failed to get cluster by ID %s", apiContext.ID)
```

Comment:
curious if this returns the error msg in API/UI as expected or requires `httperror.WrapAPIError`, could you double check?

Reply:
I'll check this

Reply:
These responses look ok to me, e.g.:
```
HTTP/1.1 500
content-encoding: gzip
content-length: 126
content-type: application/json
date: Wed, 09 Dec 2020 16:14:05 GMT
expires: Wed 24 Feb 1982 18:42:00 GMT
...

{
  "baseType": "error",
  "code": "ServerError",
  "message": "failed to get cluster by ID c-xhnqb: err",
  "status": 500,
  "type": "error"
}
```
Also, the etcd backup handler returns errors like this.